sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/_artifacts/_package/model_package.py

```diff
@@ -0,0 +1,292 @@
+        return RestPackageInputPathId(
+            input_path_type=self.input_path_type,
+            resource_id=self.resource_id,
+        )
+
+    @classmethod
+    def _from_rest_object(cls, package_input_path_id_rest_object: RestPackageInputPathId) -> "PackageInputPathId":
+        return PackageInputPathId(
+            input_path_type=package_input_path_id_rest_object.input_path_type,
+            resource_id=package_input_path_id_rest_object.resource_id,
+        )
+
+
+@experimental
+class PackageInputPathVersion:
+    def __init__(
+        self,
+        input_path_type: Optional[str] = None,
+        resource_name: Optional[str] = None,
+        resource_version: Optional[str] = None,
+    ):
+        """PackageInputPathVersion.
+
+        :param input_path_type: The type of the input path.
+        :type input_path_type: str
+        :param resource_name: The resource name of the input path.
+        :type resource_name: str
+        :param resource_version: The resource version of the input path.
+        :type resource_version: str
+        """
+        self.input_path_type = input_path_type
+        self.resource_name = resource_name
+        self.resource_version = resource_version
+
+    def _to_rest_object(self) -> RestPackageInputPathVersion:
+        return RestPackageInputPathVersion(
+            input_path_type=self.input_path_type,
+            resource_name=self.resource_name,
+            resource_version=self.resource_version,
+        )
+
+    @classmethod
+    def _from_rest_object(
+        cls, package_input_path_version_rest_object: RestPackageInputPathVersion
+    ) -> "PackageInputPathVersion":
+        return PackageInputPathVersion(
+            input_path_type=package_input_path_version_rest_object.input_path_type,
+            resource_name=package_input_path_version_rest_object.resource_name,
+            resource_version=package_input_path_version_rest_object.resource_version,
+        )
+
+
+@experimental
+class PackageInputPathUrl:
+    def __init__(self, input_path_type: Optional[str] = None, url: Optional[str] = None):
+        """PackageInputPathUrl.
+
+        :param input_path_type: The type of the input path.
+        :type input_path_type: str
+        :param url: The url of the input path.
+        :type url: str
+        """
+        self.input_path_type = input_path_type
+        self.url = url
+
+    def _to_rest_object(self) -> RestPackageInputPathUrl:
+        return RestPackageInputPathUrl(
+            input_path_type=self.input_path_type,
+            url=self.url,
+        )
+
+    @classmethod
+    def _from_rest_object(cls, package_input_path_url_rest_object: RestPackageInputPathUrl) -> "PackageInputPathUrl":
+        return PackageInputPathUrl(
+            input_path_type=package_input_path_url_rest_object.input_path_type,
+            url=package_input_path_url_rest_object.url,
+        )
+
+
+@experimental
+class ModelPackageInput:
+    """Model package input.
```

Comment:
Why is this docstring under the class definition and outside of the init, but most of the other ones are inside the constructor?

Reply:
Actually it is in class definition, let me move others into class definition too...